### PR TITLE
Make more miscellaneous design tweaks

### PR DESCRIPTION
**Pull request by galacemiguel**
**Description**
- Makes miscellaneous design tweaks on top of https://github.com/reactph/reactjsph-website/pull/57

**Summary of Changes**
1. Animated hero background
2. Added action to hero CTA button
3. Changed meetup section heading text style
4. Changed team section heading text style
5. Other minor miscellaneous changes (refer to commits)

**How to Test**
- [ ] Verify that the hero CTA button links you to our Facebook group

**Screenshots**
> Kapture 2020-02-23 at 23 51 34
> (1, 2)

> image
> (3)

> image
> (4)

### CODE REVIEW

File: src/components/hero.js [hunk 1]
```diff
@@ -1,114 +1,251 @@
...
+    const canvas = new fabric.Canvas(canvasRef.current, {
+      width: clientWidth,
+      height: clientHeight,
+    })
+
+    return canvas
+  }
+
+  useEffect(() => {
+    const canvas = initializeCanvas()
```
Comment (franreysaycon) [inline src/components/hero.js]: put initialize canvas declaration inside of useEffect. Why didn't this trigger a lint? 
If you are to use outside variables and functions, do add them as dependencies as a practice.
This is to ensure stableness because of javascript's inherent behavior of closures.

File: src/components/hero.js [hunk 2]
```diff
@@ -1,114 +1,251 @@
...
+  if (imageRatio > canvasRatio) {
+    heroImage.scaleToHeight(canvas.height)
+  } else {
+    heroImage.scaleToWidth(canvas.width)
+  }
+
+  return heroImage
+}
+
+const rotatePath = (path, midpoint, { point, angle }) => {
```
Comment (franreysaycon) [inline src/components/hero.js]: alot of these helper functions are outside the scope of the useEffect. Either move it there or register these as callbacks.

File: src/components/hero.js [hunk 3]
```diff
@@ -1,114 +1,251 @@
-import React from "react"
+import React, { useEffect, useRef, useState } from "react"
```
Comment (franreysaycon) [inline src/components/hero.js]: I think it's for the best if you just separated the whole component. Don't clutter `hero.js`.
This would make unnecessary helper functions to be more isolated. It's harder to read.

File: src/components/hero.js [hunk 4]
```diff
@@ -1,114 +1,251 @@
...
+  const INITIAL_ANGLE = -15
+  const clipPathMidpoint = new fabric.Point(
+    500 * clipPath.getObjectScaling().scaleX,
+    581.9 * clipPath.getObjectScaling().scaleY
+  )
+  let startTimestamp = 0
+  let angle = INITIAL_ANGLE
+
+  const drawClip = timestamp => {
+    if (window.innerWidth === canvas.width) {
```
Comment (franreysaycon) [inline src/components/hero.js]: potentially dangerous that may introduce weird behavior. 
I assume you are using `useEffect` for this from below. Any layout specific behavior such as measurements should be done before browser repaint. Might cause jitters, not sure. Only a theory.

File: package.json [hunk 5]
```diff
@@ -14,6 +14,7 @@
     "@rebass/preset": "^4.0.3",
     "emotion-reset": "^2.0.2",
     "emotion-theming": "^10.0.14",
+    "fabric": "3.6.2",
```
[the remark this answers is not in the document]
Comment (galacemiguel) [reply package.json]: I needed a Canvas library so `react-spring` wasn't an option. I tried `react-konva`, but the `Fabric.js` API was more convenient to use versus `Konva` where I had to write more low-level Canvas code to accomplish the same things.

I also don't believe it has its own lifecycle on top of whatever HTML Canvas already has (and whatever Canvas library we use will have the same issue anyway). 🤔

File: src/components/hero.js (hunk 1 above)
Comment (galacemiguel) [reply src/components/hero.js]: > put initialize canvas declaration inside of useEffect.

`useEffect`? Do you mean `useCallback`?

> Why didn't this trigger a lint? If you are to use outside variables and functions, do add them as dependencies as a practice.

`initializeCanvas` is being recreated on every render so its reference inside `useEffect` here can't be ever stale.

File: src/components/hero.js (hunk 2 above)
Comment (galacemiguel) [reply src/components/hero.js]: Why though?

File: src/components/hero.js (hunk 3 above)
Comment (galacemiguel) [reply src/components/hero.js]: Good point, I'll break it up!

File: src/components/hero.js (hunk 4 above)
Comment (galacemiguel) [reply src/components/hero.js]: This is being called independently of `useEffect` and [is guaranteed to run before a repaint](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame).

File: src/components/hero.js (hunk 3 above)
Comment (galacemiguel) [reply src/components/hero.js]: Done here: https://github.com/reactph/reactjsph-website/pull/58/commits/94611073cf6840332ce4420706f91f65d3ef42c8.

**Comment by galacemiguel** [comment]
Found the cause of the jittering. Will fix tomorrow. 💪

File: src/components/heroCanvas.js [hunk 6]
```diff
@@ -91,17 +91,18 @@ const drawHeroBg = (canvas, imageEl) => {
...
   const canvasContainerRef = useRef(null)
-  const canvasRef = useRef(null)
   const [imageEl, setImageEl] = useState(null)
 
   const initializeCanvas = () => {
     const { clientWidth, clientHeight } = canvasContainerRef.current
-    const canvas = new fabric.Canvas(canvasRef.current, {
+    const canvas = document.createElement("canvas")
+    canvasContainerRef.current.innerHTML = ""
+    canvasContainerRef.current.appendChild(canvas)
```
Comment (galacemiguel) [inline src/components/heroCanvas.js]: Append to and clear the canvas container imperatively since `new fabric.Canvas()` adds elements beyond just the HTML Canvas, which need to be cleared on initialization after resizing.

File: src/components/heroCanvas.js [hunk 7]
```diff
@@ -48,8 +48,15 @@ const drawRotatingBrandmark = (canvas, heroImageTemplate) => {
...
+  const initialWindowWidth = window.innerWidth
+  const initialWindowHeight = window.innerHeight
+
+  const didWindowResize = window =>
+    window.innerWidth !== initialWindowWidth ||
+    window.innerHeight !== initialWindowHeight
+
   const drawClip = timestamp => {
-    if (window.innerWidth === canvas.width) {
+    if (!didWindowResize(window)) {
```
Comment (galacemiguel) [inline src/components/heroCanvas.js]: Need to take into account if the height changed as well.

File: src/components/heroCanvas.js [hunk 8]
```diff
@@ -58,8 +58,6 @@ const drawRotatingBrandmark = (canvas, heroImageTemplate) => {
   const drawClip = timestamp => {
     if (!didWindowResize(window)) {
       fabric.util.requestAnimFrame(drawClip)
-    } else {
-      canvas.clear()
```
Comment (galacemiguel) [inline src/components/heroCanvas.js]: Don't bother clearing the old canvas since (1) it only clears the background image and not the rotating brandmark and (2) it won't animate any further anyway and should be garbage-collected soon after.

File: src/components/hero.js (hunk 2 above)
Comment (franreysaycon) [reply src/components/hero.js]: https://github.com/facebook/react/issues/14920 it's exhaustive depths. Avoid closure pitfalls. why separate it if it won't be reused by other functions anyway?

Comment (galacemiguel) [reply src/components/hero.js]: This is only an issue for functions that reference variables within the component scope. Everything is being passed in as parameters in this case, so it shouldn't be an issue! This is actually [one recommended approach to this problem](https://overreacted.io/a-complete-guide-to-useeffect/).

> First of all, if a function doesn’t use anything from the component scope, you can hoist it outside the component and then freely use it inside your effects:
>
> image

Regarding the separation, the functions were getting long and I didn't want to burden the component with the length of the code since they shouldn't be concerned with the implementation of the functions anyway.

Comment (franreysaycon) [reply src/components/hero.js]: alright. i didn't notice it sooner that it's not inside the component. If I were you, I would create a helper lib to make it easy to do unit testing but shouldn't be a big deal.

File: src/components/hero.js (hunk 4 above)
Comment (franreysaycon) [reply src/components/hero.js]: useEffect itself asynchronously run after browser paint. so any code inside that effect will only be scheduled so. So your animation although will guarantee to run before a repaint, it will only be called when useEffect is next on the event loop (the updated view was committed already though). This is the possible different "lifecycle" I was pertaining to. Though I'm not 100% sure, as long as it doesn't do any weird behavior, just taking into mind the jitters I saw back then which I think was caused by a different problem making this concern invalid, I'm fine with it. =)) Let's resolve this.
